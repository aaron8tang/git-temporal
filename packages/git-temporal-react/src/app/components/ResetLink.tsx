import React from 'react';
import { style } from 'app/styles';
import { UndoIcon } from './UndoIcon';

export interface ResetLinkProps {
  // This is the text or JSX that gets wrapped in stacked label
  children: string | JSX.Element | JSX.Element[];
  onClick: (evt) => void;
  title?: string;
  style?: object;
}

const defaultContainerStyle = {
  _extends: ['smallerText', 'link', 'inlineBlock'],
  marginRight: '@margins.pageRight+px',
  marginLeft: '@margins.medium+px',
};

export const ResetLink = (props: ResetLinkProps): JSX.Element => {
  return (
    <div
      style={style(defaultContainerStyle, props.style)}
      onClick={props.onClick}
    >
      <UndoIcon height={10} width={10} /> {props.children}
    </div>
  );
};

ResetLink.displayName = 'ResetLink';
