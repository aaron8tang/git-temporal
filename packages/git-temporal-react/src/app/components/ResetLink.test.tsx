import React from 'react';
import { shallow } from 'enzyme';

import { ResetLink } from './ResetLink';

describe('components/ResetLink', () => {
  describe('when rendered with required props', () => {
    let wrapper;
    const onClick = jest.fn();
    beforeAll(() => {
      wrapper = shallow(
        <ResetLink onClick={onClick}>this is the text</ResetLink>
      );
    });

    test('it should match snapshot (it should have rendered a label and a text element)', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
