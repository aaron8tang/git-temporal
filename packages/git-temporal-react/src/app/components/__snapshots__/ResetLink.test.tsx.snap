// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ResetLink when rendered with required props it should match snapshot (it should have rendered a label and a text element) 1`] = `
<div
  onClick={[MockFunction]}
  style={
    Object {
      "color": "blue",
      "cursor": "pointer",
      "display": "inline-block",
      "fontSize": 11,
      "fontWeight": "normal",
      "lineHeight": 1.35,
      "marginLeft": "7px",
      "marginRight": "20px",
      "textDecoration": "underline",
    }
  }
>
  <UndoIcon
    height={10}
    width={10}
  />
   
  this is the text
</div>
`;
